tests/Integration/UI/SettingsPageTest.php

```diff
@@ -44,153 +44,185 @@ public function set_up(): void {
 		self::assertSame( $expected, $actual );
 	}
 
 	/**
-	 * Verifies that valid API Secret values are retained when validated.
+	 * Verifies that invalid API credentials are emptied when validated with the Validation API
 	 *
-	 * @since 3.9.0
+	 * @since 3.11.0
 	 *
 	 * @covers \Parsely\UI\Settings_Page::validate_basic_section
-	 * @uses \Parsely\Parsely::__construct
-	 * @uses \Parsely\Parsely::api_secret_is_set
-	 * @uses \Parsely\Parsely::are_credentials_managed
-	 * @uses \Parsely\Parsely::get_api_secret
-	 * @uses \Parsely\Parsely::get_options
-	 * @uses \Parsely\Parsely::get_managed_credentials
-	 * @uses \Parsely\UI\Settings_Page::__construct
-	 * @uses \Parsely\UI\Settings_Page::get_logo_default
-	 * @uses \Parsely\UI\Settings_Page::get_obfuscated_value
-	 * @uses \Parsely\UI\Settings_Page::get_unobfuscated_value
-	 * @uses \Parsely\UI\Settings_Page::sanitize_site_id
-	 * @uses \Parsely\UI\Settings_Page::validate_advanced_section
-	 * @uses \Parsely\UI\Settings_Page::validate_options
-	 * @uses \Parsely\UI\Settings_Page::validate_options_post_type_tracking
-	 * @uses \Parsely\UI\Settings_Page::validate_recrawl_section
-	 * @uses \Parsely\Validator::validate_api_secret
-	 * @uses \Parsely\Validator::validate_site_id
+	 * @uses   \Parsely\Parsely::__construct
+	 * @uses   \Parsely\Parsely::api_secret_is_set
+	 * @uses   \Parsely\Parsely::are_credentials_managed
+	 * @uses   \Parsely\Parsely::get_api_secret
+	 * @uses   \Parsely\Parsely::get_options
+	 * @uses   \Parsely\Parsely::get_managed_credentials
+	 * @uses   \Parsely\UI\Settings_Page::__construct
+	 * @uses   \Parsely\UI\Settings_Page::get_logo_default
+	 * @uses   \Parsely\UI\Settings_Page::get_obfuscated_value
+	 * @uses   \Parsely\UI\Settings_Page::get_unobfuscated_value
+	 * @uses   \Parsely\UI\Settings_Page::sanitize_site_id
+	 * @uses   \Parsely\UI\Settings_Page::validate_advanced_section
+	 * @uses   \Parsely\UI\Settings_Page::validate_options
+	 * @uses   \Parsely\UI\Settings_Page::validate_options_post_type_tracking
+	 * @uses   \Parsely\UI\Settings_Page::validate_recrawl_section
+	 * @uses   \Parsely\Validator::validate_api_credentials
 	 *
 	 * @group settings-page
 	 * @group settings-page-validation
 	 */
-	public function test_valid_api_secret_values_are_retained_when_validated(): void {
-		$options = self::$parsely->get_options();
+	public function test_invalid_api_credentials_are_emptied_when_validated(): void {
+		remove_filter( 'pre_http_request', array( $this, 'mock_request_api_credentials_validation_success' ), 10 );
+		// Mock HTTP request to simulate a failed credentials validation.
+		add_filter( 'pre_http_request', array( $this, 'mock_request_api_credentials_validation_failure' ), 10, 3 );
+
+		$expected = self::$parsely->get_options();
+		$options  = self::$parsely->get_options();
 
-		// More than 30 characters.
+		$options['apikey']     = 'mydomain.com';
 		$options['api_secret'] = 'valid_api_secret_key_based_on_length';
```

Comment:
```suggestion
		$options['api_secret'] = 'invalid_api_secret';
```
Although the value doesn't have any real effect in the test, let's change it so it's coherent with the test's purpose.